qualtran/resource_counting/symbolic_counting_utils.py

```diff
@@ -68,7 +77,19 @@ def smax(*args):
     return max(*args)
 
 
+def smin(*args):
+    if is_symbolic(*args):
+        return sympy.Min(*args)
+    return min(*args)
+
+
 def acos(x: SymbolicFloat) -> SymbolicFloat:
     if not isinstance(x, sympy.Basic):
         return np.arccos(x)
     return sympy.acos(x)
+
+
+def length(x: Union[Sized, Shaped]) -> SymbolicInt:
```

Comment:
I think we need a better naming convention. To someone reading qualtran code, seeing a `length` function could be very confusing: what is this function? why not `len`? 

With the min function, you named it `smin` which I think is a good idea to disambiguate it from `min`. maybe `slen`. Another alternative would be to switch to using a namespaced name at the call site:

```python
import qualtran.resource_counting.symbolic_counting_utils as qlt_symb

n = qlt_symb.length(xxx)
```

but of course we'd still have to use names that don't clash with builtin `min`, `len`; so maybe the former convention is better.

Reply:
used `slen` to match the existing symbolic functions.